src/passwordValidation.js

```diff
@@ -10,22 +10,45 @@ export default function isValidPassword(password = "") {
   // The following line ensures, that password is always a string, like the number 128 -> string "128"
   if (typeof password !== "string") password = String(password);
 
-  if(password.length !== 10){
-    return false
-  }
+  if (password.length !== 10) return false;
 
-  if (/[^a-zA-Z0-9]/.test(password)) {
-    return false;
-  }
+  // The password that only has numbers is invalid
+  if (/^[0-9]*$/.test(password)) return false;
 
- if(/^[A-Za-z0-9].{10}$/.test(password)){
-  return false
- }
+  // The password that only has characters is invalid
+  if (/^[A-Za-z]*$/.test(password)) return false;
 
- if(/^[A-Za-z0-9]*$/.test(password)){
-  return false
- }
+  // a password with special characters is invalid
+  if (/[^a-zA-Z0-9]/.test(password)) return false;
 
+  // a password with only lower or upper case characters is invalid
+  if (password === password.toLowerCase() || password === password.toUpperCase()) return false;
+
+  //  forbidden password are invalid
+  if (forbiddenPasswords.includes(password)) return false;
```

Comment:
This test should be conducted before multiple predictive tests. There is no need for an invalid password to go through multiple tests if it is already included in the list of forbidden passwords